### PR TITLE
Feat/reset password page

### LINKED ISSUE #140: Add fronted functionality for 'forgot password'
When a user forgets his password he needs to be able to reset it without logging in. While there is a backend endpoint for it, it also needs to be implemented in the frontend. 
There already exists a link for it, it just has to link to a page to enter an email and then send a backend request to reset the password.
Screenshot 2022-07-05 at 18 04 25

See 'forgot password?' link under the form

### PR DESCRIPTION
**Pull request by codingfabi**
Closes #140 .

Right now the functionality is restricted because the reset email is not designed as it should be (#150). But there are no more code changes required in the frontend.

### CODE REVIEW

**Review by sami1riaz — CHANGES_REQUESTED** [review]

File: frontend/pages/set-new-password.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  })
+
+  const handleDialogClose = () => {
+    resetPasswordForm.resetForm();
+    setErrorState(false)
+  }
+
+
+  return (
+    <>
```
[the remark this answers is not in the document]
Comment (codingfabi) [reply frontend/pages/set-new-password.tsx:84]: Yes, I like to wrap all html elements in to a react fragment. I changed the shortcut to the actual fragment element to clear this up. I would argue that this does not pollute the code but rather adds some structure.

File: frontend/pages/reset-password.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+import withRoot from '../src/withRoot';
+import Container from '@material-ui/core/Container';
+
+import { useState } from 'react';
+
+import { gql, useMutation } from '@apollo/client'
+import { useRouter } from 'next/router'
+
+
+const RESET_PASSWORD = gql`
```
[the remark this answers is not in the document]
Comment (codingfabi) [reply frontend/pages/reset-password.tsx:18]: renamed

File: frontend/pages/reset-password.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
+import { Button } from '@material-ui/core';
```
[the remark this answers is not in the document]
Comment (codingfabi) [reply frontend/pages/reset-password.tsx]: I agree but would argue that this has not really a high priority at the moment. Raising an issue is important but I think we can take care of it later.

File: frontend/pages/set-new-password.tsx [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+      <AppAppBar />
+      <PageContainer title="Set New Password">
+      <div style={{ padding: "48px 16px" }}>
+        <Container maxWidth="xs">
+                  <form onSubmit={resetPasswordForm.handleSubmit}>
+                  <TextField
+                  id="resetToken"
```
[the remark this answers is not in the document]
Comment (codingfabi) [reply frontend/pages/set-new-password.tsx:95]: I agree that this workflow is a bit unconventional and inconvenient and we could also do this with the token in a url parameter.  but right now we do not have figured a concept on creating the url structure whether the application is deployed in a productive, a testing or a local environment. To implement this feature ASAP I would argue to handle it the way I implemented it for now and restructure the approach later. 

The same applies for the confirmation of a new account.

[the remark this answers is not in the document]
Comment (sami1riaz) [reply frontend/pages/set-new-password.tsx:95]: yeah I agree, it works for now

File: frontend/pages/set-new-password.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (sami1riaz) [reply frontend/pages/set-new-password.tsx]: Ah, I didn't realize that's what it was. Good to know

File: frontend/pages/set-new-password.tsx [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+                  />
+                  <TextField 
+                    id="newPasswordRepeat"
+                    label="Repeat New Password"
+                    variant="outlined"
+                    fullWidth
+                    required
+                    type="password"
+                    value={resetPasswordForm.values.newPasswordRepeat}
+                    onChange={resetPasswordForm.handleChange}
```
Comment (sami1riaz) [inline frontend/pages/set-new-password.tsx:129]: These text fields don't have the error attribute, so they won't display the formik errors for min number of chars and same password

**Comment by sami1riaz** [comment]
Left one more comment but other than that the changes look good. I did notice that the mutation is a success regardless of what email we enter (as long as it is the correct format). Tried with my non-registered email and the response from the mutation was a success and I was redirected to the home page, but I didn't receive an email.

File: frontend/pages/set-new-password.tsx (hunk 5 above)
Comment (codingfabi) [reply frontend/pages/set-new-password.tsx:131]: added

**Comment by codingfabi** [comment]
I opened an issue for this problem(#164) but I think this should be taken care of in a different PR.

**Pull request merged**